firmware/repository.py

```diff
@@ -0,0 +1,111 @@
+import sqlite3
+import os
+from flask import Flask, request, session, g, redirect, url_for, abort, render_template, flash
+
+from uploaders import upload_file
+from firmware import app
+
+import pdb
+
+def connect_db():
+    rv = sqlite3.connect(app.config['DATABASE'], timeout=1)
+    rv.row_factory = sqlite3.Row
+    return rv
+
+
+def init_db():
+    db = get_db()
+    with app.open_resource('schema.sql', mode='r') as f:
+        db.cursor().executescript(f.read())
+    db.commit()
+
+
+@app.cli.command('initdb')
+def initdb_command():
+    init_db()
+    print("Firmware initialized database sucessfully!")
+
+
+def get_db():
+    if not hasattr(g, 'sqlite_db'):
+        g.sqlite_db = connect_db()
+    return g.sqlite_db
+
+
+@app.teardown_appcontext
+def close_db(error):
+    if hasattr(g, 'sqlite_db'):
+        g.sqlite_db.close()
+
+
+def get_companies():
+    db = get_db()
+    cur = db.execute("select * from companies")
+    companies = cur.fetchall()
+    return companies
+
+
+def add_company(request_form, request_files):
+    db = get_db()
+    add_cursor = db.execute("select id from categories where \
+        type='%s'" % str(request_form.get('select-category-list')))
+    category_id = add_cursor.fetchall()[0]
```

Comment:
You should find a better name for `add_cursor`. Also at this point, you should already have the ID of the category.

Reply:
Changed names.Last part to do.